morango/utils/syncing_utils.py

```diff
@@ -1,14 +1,93 @@
     for model_class in django.apps.apps.get_models():
+        # several validation checks to assert models will be syncing correctly
         if issubclass(model_class, SyncableModel) and not model_class._meta.proxy:
-            model_list.append(model_class)
+            name = model_class.__name__
+            try:
+                from mptt import models
+                from morango.utils.morango_mptt import MorangoMPTTModel, MorangoMPTTTreeManager, MorangoTreeQuerySet
+                # mptt syncable model checks
+                if issubclass(model_class, models.MPTTModel):
+                    if not issubclass(model_class, MorangoMPTTModel):
+                        raise InvalidMorangoModelConfiguration("{} that inherits from MPTTModel, should instead inherit from MorangoMPTTModel.".format(name))
+                    if not isinstance(model_class.objects, MorangoMPTTTreeManager):
+                        raise InvalidMPTTManager("Manager for {} must inherit from MorangoMPTTTreeManager.".format(name))
+                    if not isinstance(model_class.objects.none(), MorangoTreeQuerySet):
+                        raise InvalidMPTTQuerySet("Queryset for {} model must inherit from MorangoTreeQuerySet.".format(name))
+            except ImportError:
+                pass
+            # syncable model checks
+            if not isinstance(model_class.objects, SyncableModelManager):
+                raise InvalidSyncableManager("Manager for {} must inherit from SyncableModelManager.".format(name))
+            if not isinstance(model_class.objects.none(), SyncableModelQuerySet):
+                raise InvalidSyncableQueryset("Queryset for {} model must inherit from SyncableModelQuerySet.".format(name))
+            if model_class._meta.many_to_many:
+                raise UnsupportedFieldType("{} model with a ManyToManyField is not supported in morango.")
+            if not hasattr(model_class, 'morango_model_name'):
+                raise InvalidMorangoModelConfiguration("{} model must define a morango_model_name".format(name))
+
+            # create empty list to hold model classes for profile if not yet created
+            profile = model_class._morango_profile
+            _profiles_for_syncing_models[profile] = _profiles_for_syncing_models.get(profile, [])
+
+            # special case for root proxy model
+            if model_class._meta.proxied_children:
+                model_class._meta.proxied_children.reverse()
+                for proxy_model in model_class._meta.proxied_children:
+                    _insert_model_into_profile_dict(proxy_model.model, profile)
+            else:
+                _insert_model_into_profile_dict(model_class, profile)
 
-    for model_class in model_list:
-        _syncing_models.update({model_class.morango_model_name: model_class})
+    # for each profile, create a dict mapping from morango model names to model class
+    for profile, model_list in iteritems(_profiles_for_syncing_models):
```

Comment:
The more I read the variable name `_profiles_for_syncing_models`, the less it means to me. Anything more intuitive we can find?